### DISCUSSION TITLE
add non-interactive mode for soup

### DISCUSSION THREAD

**Discussion by Ron89** [Security-Onion-Solutions/securityonion#5005] [Ideas] [answered]
The maintenance of our deployment of Security Onion is mostly done remotely through ssh. When the internet connection becomes unstable, or when there is a long inactive period, the ssh session may get terminated. Since soup takes a long time, if we have to operate soup in an interactive manner, when ssh session gets terminated / broken, the soup process may get killed at random timing, and that is quite scary.

If I can start soup in an non-interactive way, then I can use `nohup` to start soup and detach it from my ssh session, so that no matter when my ssh session gets terminated, the `soup` process won't be affected. The only problem is, there is a compulsory interaction at the beginning of the `soup` process. So if I start soup in a detached manner, I won't be able to send in the confirmation. Though it is possible for me to send in confirmation then detach soup from my ssh session. But it is hacky and if something goes wrong when I detach it, there is still a chance that something might get corrupted.

Is it possible to add a command line argument, so that the confirmation step can be skipped? Specifically, putting this line
https://github.com/Security-Onion-Solutions/securityonion/blob/837c0402a0098ecdc691643787496dd439427c74/salt/common/tools/sbin/soup#L1011 into a if clause?

**Comment by TOoSmOotH** [comment]
I use screen or tmux for occasions like this.

**Reply by Ron89** [reply]
Good point. Tmux does help. But may I know the rationale of making interaction compulsory for `soup` workflow?

**Comment by dougburks** [comment, marked as answer]
The `read -r input` on line 1011 is not the only part of soup that requires interaction, so it will be a little more involved but we'll take a look as time allows:
https://github.com/Security-Onion-Solutions/securityonion/issues/5043
